### PR TITLE
Re-try API requests if Slack API rate limit has been reached.

### PR DESCRIPTION
**Pull request by skiptomyliu**
It's easy to hit ratelimits with large orgs, especially when paginating users.  This addition will retry requests every 10s for 1 minute.

### CODE REVIEW

File: securitybot/chat/slack.py [hunk 1]
```diff
@@ -37,20 +41,33 @@ def _validate(self):
...
             (dict): Parsed JSON from the response.
         '''
-        response = self._slack.api_call(method, **kwargs)
+        cur_try = 0
+        while True:
+            response = self._slack.api_call(method, **kwargs)
+            if cur_try > RATE_LIMIT_TRIES:
+                raise ChatException('Slack rate limit max tries reached.')
+
+            if response.get('error', '').lower() == 'ratelimited' and rate_limit_retry is True:
```
Comment (42B) [inline securitybot/chat/slack.py]:
```python
... and rate_limit_retry:  # ?
```

> Don’t compare boolean values to True or False using ==:
> 
> Yes:
> 
>     if greeting:
> 
> No:
> 
>     if greeting == True:
> 
> Worse:
> 
>     if greeting is True:

* https://pep8.org/#programming-recommendations

File: securitybot/chat/slack.py [hunk 2]
```diff
@@ -84,7 +101,19 @@ def get_users(self):
                     }
             }
         '''
-        return self._api_call('users.list')['members']
+        members = []
+        next_cursor = None
+        while True:
+            response = self._api_call('users.list', cursor=next_cursor)
+            active_members = [m for m in response['members'] if m.get('deleted') is False]
```
Comment (42B) [inline securitybot/chat/slack.py]:
```python
... if not m.get('deleted')]  # ?
```

> Don’t compare boolean values to True or False using ==:
> 
> Yes:
> 
>     if greeting:
> 
> No:
> 
>     if greeting == True:
> 
> Worse:
> 
>     if greeting is True:

* https://pep8.org/#programming-recommendations

File: securitybot/chat/slack.py (hunk 1 above)
Comment (skiptomyliu) [reply securitybot/chat/slack.py]: Thank you.  I'm making the corrections.

**Comment by skiptomyliu** [comment]
Sorry, may have jumped gun -- cursor is not being set properly.  Will reopen later.

**Pull request closed without merging**